### PR TITLE
Type2 uncertainty added

### PR DESCRIPTION
**Pull request by jpa198**

### CODE REVIEW

File: SrfGen/gcmt2srf.py [hunk 1]
```diff
@@ -82,100 +87,210 @@ def run_create_srf(args, t, vs, rho, n_sims):
...
+    )
+    uncertainty_group.add_argument(
+        "-c", "--cybershake-fault-file", dest="cs_file", type=str, default=""
+    )
+    args, extra_args = parser.parse_known_args()
+    if args.type == 1:
+        pass
+    elif args.type == 2:
+        psff_parser = ArgumentParser(parents=[parser])
+        psff_parser.add_argument(
```
Comment (jasonmotha) [inline SrfGen/gcmt2srf.py]: default?

File: SrfGen/gcmt2srf.py [hunk 2]
```diff
@@ -82,100 +87,210 @@ def run_create_srf(args, t, vs, rho, n_sims):
...
-if any_opts and not all_opts:
-    parser.error(
-        "If realisation uncertainty or a cybershake fault file is given then the other must be given also."
+
+def main():
+    # load parameters
+    parser = ArgumentParser(add_help=False)
+    parser.add_argument("csv_file", help="path to CMT solutions CSV")
+    parser.add_argument("velocity_model", help="path to 1D velocity model")
+    parser.add_argument(
```
Comment (jasonmotha) [inline SrfGen/gcmt2srf.py]: I don't think this is the functionality we want. @lee-robin Should it be FF for mw > 5.5 or how do we want to isolate doing type 2 generation vs type 1 generation.

File: SrfGen/createSRF.py [hunk 3]
```diff
@@ -671,8 +684,128 @@ def CreateSRF_ff(
...
+    slip_cov=None,
+    silent=False,
+):
+    """
+    Create a Finite Fault SRF.
+    Calculates flen, dlen... if not supplied given depth and mwsr are keywords.
+    NOTE: depth is focal mech depth and only used when calculating flen etc...
+    """
+
+    seed_file = "{}.SEED".format(prefix)
```
Comment (jasonmotha) [inline SrfGen/createSRF.py]: Extract to function and create comment. As we will use this for Type 2, 3, 4

File: SrfGen/createSRF.py [hunk 4]
```diff
@@ -671,8 +684,128 @@ def CreateSRF_ff(
...
+        ny,
+        seed,
+        shypo,
+        dhypo,
+        genslip_version=genslip_version,
+        rvfrac=rvfrac,
+        slip_cov=slip_cov,
+        rough=rough,
+    )
+    with open(seed_file, "w") as sf:
```
Comment (jasonmotha) [inline SrfGen/createSRF.py]: Function again

**Comment by jpa198** [comment]
Not likely to get back to this till after AM

File: SrfGen/gcmt2srf.py (hunk 1 above)
Comment (jpa198) [reply SrfGen/gcmt2srf.py]: Not sure what you meant by this?

**Pull request merged**